src/main/java/org/folio/services/lines/PoLineNumbersService.java

```diff
@@ -1,38 +1,55 @@
 package org.folio.services.lines;
 
 import io.vertx.core.Future;
-import org.folio.dao.PostgresClientFactory;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.folio.rest.jaxrs.model.PoLineNumber;
+import org.folio.rest.jaxrs.model.PurchaseOrder;
+import org.folio.rest.persist.DBClient;
 import org.folio.rest.persist.PostgresClient;
-import org.folio.services.order.OrderSequenceRequestBuilder;
 
-import java.util.ArrayList;
 import java.util.List;
+import java.util.stream.IntStream;
+
+import static java.util.stream.Collectors.toList;
+import static org.folio.models.TableNames.PURCHASE_ORDER_TABLE;
 
 public class PoLineNumbersService {
-  private OrderSequenceRequestBuilder orderSequenceBuilder;
-  private PostgresClientFactory pgClientFactory;
+  private static final Logger log = LogManager.getLogger();
+
+  private final PoLinesService poLinesService;
 
-  public PoLineNumbersService(OrderSequenceRequestBuilder orderSequenceBuilder, PostgresClientFactory pgClientFactory) {
-    this.orderSequenceBuilder = orderSequenceBuilder;
-    this.pgClientFactory = pgClientFactory;
+  public PoLineNumbersService(PoLinesService poLinesService) {
+    this.poLinesService = poLinesService;
   }
 
   /**
-   * Retrieves next po line number from sequence for order.
-   *
-   * @param purchaseOrderId the order id
-   * @param poLineNumbers the po line numbers
-   * @param tenantId the tenant id
-   * @return future with po line number
+   * Returns poLineNumbers new po line numbers for the order.
    */
-  public Future<PoLineNumber> retrievePoLineNumber(String purchaseOrderId, int poLineNumbers, String tenantId) {
-    PostgresClient pgClient = pgClientFactory.createInstance(tenantId);
-    return pgClient.select(orderSequenceBuilder.buildPOLNumberQuery(purchaseOrderId, poLineNumbers))
-      .map(results -> {
-        List<String> sequenceNumbers = new ArrayList<>();
-        results.forEach(row -> sequenceNumbers.add(row.getLong(0).toString()));
+  public Future<PoLineNumber> retrievePoLineNumber(String purchaseOrderId, int poLineNumbers, DBClient dbClient) {
+    PostgresClient pgClient = dbClient.getPgClient();
+    log.debug("retrievePoLineNumber: getting po {} for update", purchaseOrderId);
+    return pgClient.withConn(conn -> conn.getByIdForUpdate(PURCHASE_ORDER_TABLE, purchaseOrderId, PurchaseOrder.class)
```

Comment:
It would be better to move _conn.getByIdForUpdate(PURCHASE_ORDER_TABLE, purchaseOrderId, PurchaseOrder.class)_
to dao layer.

Reply:
I created a new DAO for orders. So far DB calls had not been using a DAO layer for orders, this is probably something we could improve.